### PR TITLE
Handle manifest.json containing null in tarball

### PR DESCRIPTION
**Pull request by knqyf263**
If the content of manifest.json is null, it causes panic. I added the tar file causing panic. You can test it by `TestNullManifest`.

Also, this is PoC.
https://play.golang.org/p/RVdS6KY40-m

I faced this problem when I used Kaniko.

```
$ cat Dockerfile
FROM alpine:3.11
$ docker run -v $PWD:/workspace gcr.io/kaniko-project/executor:latest --tarPath /workspace/image.tar --dockerfile /workspace/Dockerfile --context dir:///workspace/ --no-push
```

I know this is a bug of Kaniko and it doesn't generate null manifest if we add `--destination`.

```
$ docker run -v $PWD:/workspace gcr.io/kaniko-project/executor:latest --tarPath /workspace/image.tar --dockerfile /workspace/Dockerfile --context dir:///workspace/ --no-push --destination=image
```

But I feel this library should handle this panic.

### CODE REVIEW

File: pkg/v1/tarball/image.go [hunk 1]
```diff
@@ -162,6 +162,10 @@ func (i *image) loadTarDescriptorAndConfig() error {
 		return err
 	}
 
+	if i.manifest == nil{
```
Comment (jonjohnsonjr) [inline pkg/v1/tarball/image.go]:
```suggestion
	if i.manifest == nil {
```

File: pkg/v1/tarball/image_test.go [hunk 2]
```diff
@@ -47,6 +47,13 @@ func TestManifestAndConfig(t *testing.T) {
 	}
 }
 
+func TestNullManifest(t *testing.T) {
+	img, err := ImageFromPath("testdata/null_manifest.tar", nil)
+	if err == nil {
+		t.Fatalf("Error expected loadig image: %v", img)
```
Comment (jonjohnsonjr) [inline pkg/v1/tarball/image_test.go]:
```suggestion
		t.Fatalf("Error expected loading null image: %v", img)
```

File: pkg/v1/tarball/image.go (hunk 1 above)
Comment (knqyf263) [reply pkg/v1/tarball/image.go]: Thanks. My linter didn't work...I've fixed it.

File: pkg/v1/tarball/image_test.go (hunk 2 above)
Comment (knqyf263) [reply pkg/v1/tarball/image_test.go]: fixed

**Comment by jonjohnsonjr** [comment]
Thanks!

**Pull request merged**